app/src/pages/[resourceType]/[id].tsx

```diff
@@ -29,14 +36,49 @@ export default function ResourceIDPage({ jsonData }: InferGetServerSidePropsType
     return decode('application/elm+json', jsonData);
   }, [jsonData]);
 
+  const ctx = trpc.useContext();
+
+  const draftMutation = trpc.draft.createDraft.useMutation({
+    onSuccess: data => {
+      notifications.show({
+        title: `${jsonData.resourceType} Created!`,
+        message: `${jsonData.resourceType} successfully created from ${jsonData.resourceType}/${jsonData.id}`,
+        icon: <CircleCheck />,
+        color: 'green'
+      });
+      router.push(`/authoring/${jsonData.resourceType}/${data.draftId}`);
+      ctx.draft.getDraftCounts.invalidate();
+    },
+    onError: e => {
+      notifications.show({
+        title: `${jsonData.resourceType} Creation Failed!`,
+        message: `Attempt to create ${jsonData.resourceType} from ${jsonData.resourceType}/${jsonData.id} failed with message: ${e.message}`,
+        icon: <AlertCircle />,
+        color: 'red'
+      });
+    }
+  });
+
+  const router = useRouter();
+
+  const createDraftOfArtifact = () => {
+    const draftOfArtifact = jsonData;
+    draftOfArtifact.id = uuidv4();
+    draftOfArtifact.status = 'draft';
+    draftMutation.mutate({ resourceType, draft: draftOfArtifact });
+  };
```

Comment:
Hmm, good suggestion that I didn't think about. I am leaning toward keeping `createDraft` as is. I think it's best if `createDraft` is kept as simple as possible so that we can reuse it in multiple places

Reply:
I did notice that we have the same code in `app/src/service/trpc/routers/service` and `app/src/pages/[resourceType]/[id].tsx` for setting the id, status, and deleting the version, so maybe a compromise here would be to create a helper function under the `util` directory called `modifyResourceFields` (or something similar) that takes care of these modifications. But we aren't duplicating that much code so it's not a big deal